### ISSUE TITLE
Unable to mount or read file/folder in system mounted partition file system

### ISSUE THREAD

**Issue by yoroshikun** [craftcms/nitro#210]
### Description
I have started to partition my disk into different sections so that I can keep my home and developer files separated from each other, however this seems to cause issues with nitro.

I am receiving a does not exist error when attempting to add site that is located in a mounted partition.

This also appears when trying to import a database using x.sql that is located on a mounted partition (not home / nitro install partition) the importing fails with 
```Source Path "/<mountpoint>/path/to/sql/x.sql" does not exist
Error: exit status 1
```

I am new to linux, any insight would be helpful :)

#### Adding Site
1. Setup Nitro as per [installation instructions](https://craftcms.com/docs/nitro/installation.html)
2. Attempt to add project with nitro add (or editing nitro-dev.yaml manually and nitro apply)
3. Returns error, 
```Mounting /<mountpoint>/path/to/site
Adding site <sitename> to nitro-dev
Source path "/<mountpoint>/path/to/site" does not exist
Error: exit status 1
```

#### Importing DB
1. Setup Nitro as per [installation instructions](https://craftcms.com/docs/nitro/installation.html)
2. Create database with `nitro db add`
3. Attempt import of database with `nitro db import <database>.sql`
4. Returns error,
```Uploading "<database>.sql" into "nitro-dev" (large files may take a while)...
Source path "/<mountpoint>/path/to/site/<database>.sql" does not exist
Error: exit status 1
```

### Additional info

- OS: Linux Mint 20 Cinnamon
- Nitro version: 1.0.1
- Multipass version: 1.4.0

Drive is partitioned with two non system partitions, home and share (<mountpoint>)
Home, /dev/sda3
share, /dev/sda4

All fs are formated with ext4

#### Workaround

Everything works fine if I were to run the `nitro add` and `import` commands from the home /dev/sda3 part of the file system.

**Comment by yoroshikun** [comment]
I have done some more research into this and I believe the information I have gathered may be of use to anyone in a similar situation to me.

#### The Issue
The issue stems from the fact that multipass (installed though snap?) does not have access to files outside of the $HOME directories. Which was commented on [Here](https://forum.snapcraft.io/t/after-upgraded-multipass-to-1-3-0-snapcraft-cannot-work-any-more/18089/4)

#### The solution / Workaround
Also in the same post, offered a solution of bind-mounting. The concept being new to me I did more research to determine, how and if its safe.
[Is it safe to mount the same device twice](https://www.reddit.com/r/linuxquestions/comments/bh3xxh/is_it_okay_to_mount_the_same_device_twice/) 
[What is a bind mount](https://unix.stackexchange.com/questions/198590/what-is-a-bind-mount)

From here I decided it  was my best bet to create a Read Write copy of the `/share` directory (mounted from `/dev/sda4`) and bind mount it under `$HOME/share` this way multipass could access it.

#### TLDR
Use [Bind Mounts](https://unix.stackexchange.com/questions/198590/what-is-a-bind-mount) to remount a copy of the file system under the $HOME directory.

```bash
mount SRC DST -o bind,rw
```
in older linux version you may need to write this out in two lines (util-linux <=2.26)
```bash
mount SRC DST -o bind
mount DST -o remount,rw,bind
```
[Why these commands?](https://unix.stackexchange.com/questions/128336/why-doesnt-mount-respect-the-read-only-option-for-bind-mounts)

#### More Information
I encountered a different error while getting craft to work on my linux install which I have created a new Issue detailing it, and the workaround that I am using for now.
The issue is here #213

**Comment by jasonmccallister** [comment]
@yoroshikun first of amazing work and research, I just finished reading the articles. That is one downside to the multipass installation over snap and that had not occurred to  me until I read your research.

On my Ubuntu machine, I have an multiple SSDs mounted, but they end up under `/home/jasonmccallister/dev` so I was not seeing the issue. I also did not have multipass installed through snap as I switch between unstable and stable builds. 

Maybe this makes more sense as a documentation on _how_ to mount drives that are outside of permissions from snap? I wonder if flatpak has the same issue.

Thanks again!

**Comment by jasonmccallister** [comment]
We just released Nitro 2 where we switched from Multipass to Docker and added a ton of new features. Please see the [announcement for more details](https://craftcms.com/blog/nitro-2).

This issue was specific to Multipass so we are going to close this out. Please see the documentation on upgrading from [Nitro 1](https://craftcms.com/docs/nitro/2.x/upgrade.html).
